src/dataflow/components/dataflow-program.tsx

```diff
@@ -220,6 +223,36 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
   private setProgramRunTime = (time: number) => {
     this.setState({programRunTime: time});
   }
+  private generateProgramData = () => {
+    let programName: any = "my-program";
+    const endTimePad = 24 * 60 * 60 * 1000;
+    const newTimestamp = Date.now() + endTimePad;
+    const hubs: string[] = [];
+    const sensors: string[] = [];
+    this.programEditor.nodes.forEach((n: Node) => {
+      if (n.name === "Sensor" && n.data.sensor) {
+        const chInfo = this.channels.find(ci => ci.channelId === n.data.sensor);
+        if (chInfo) {
+          hubs.push(chInfo.hubId);
+          sensors.push(`${chInfo.hubId}_${chInfo.channelId}`);
+        }
+      } else if (n.name === "Data Storage" && n.data.datasetName) {
+        programName = n.data.datasetName;
+      }
+    });
+    const programData = {
+      program: {
+        endTime: newTimestamp,
+        hubs,
+        // tslint:disable-next-line:max-line-length
+        program: "\"{\"file_version\":\"0.1\",\"blocks\":[{\"type\":\"sensor\",\"source\":\"2711-tempe\",\"save\":true,\"label\":\"My temperature\"},{\"type\":\"sensor\",\"source\":\"2711-humid\",\"save\":true,\"label\":\"My humidity\"}]}\"",
+        programId: programName ? programName : "test",
+        runInterval: 2000,
```

Comment:
@dstrawberrygirl is this the record interval used to set how often we write to a dataset (record data)?  Want to eventually get this off of the data storage block.

Reply:
Yes this needs to be from the data storage block

Reply:
And we should also make the programName mandatory - it should be a unique identifier so that a user can retrieve their specific program. Perhaps the uniqueness is stored as the combination of name and timestamp (since ms clashes are unlikely) but perhaps we need to be more specific about a program name plus a run number or similar.

Reply:
Will add.

Reply:
Added the interval from the data storage block.  Didn't touch the program name.